src/components/media-views.js

```diff
@@ -837,3 +846,150 @@ AFRAME.registerComponent("media-image", {
+
+  init() {},
+
+  remove() {
+    if (this.data.batch && this.mesh) {
+      this.el.sceneEl.systems["hubs-systems"].batchManagerSystem.removeObject(this.mesh);
+    }
+
+    if (this.currentPageTextureIsRetained) {
+      const cacheKey = this.currentTextureCacheKey();
+      textureCache.release(cacheKey);
+      this.currentPageTextureIsRetained = false;
+    }
+  },
+
+  currentTextureCacheKey() {
+    // We ensure a unique texture for each PDF entity, because they are drawn over during pagination.
+    return `${this.el.object3D.uuid}_${this.canvas.width}_${this.canvas.height}`;
+  },
+
+  async update(oldData) {
+    let texture;
+    let ratio = 1;
+
+    try {
+      const { src, index } = this.data;
+      if (!src) return;
+
+      this.el.emit("pdf-loading");
+
+      if (!oldData.src || src !== oldData.src) {
+        const loadingSrc = this.data.src;
+        this.pdf = await pdfjs.getDocument(src);
+        if (loadingSrc !== this.data.src) return;
+        this.el.setAttribute("media-pager", { maxIndex: this.pdf.numPages - 1 });
+      }
+
+      const loadingIndex = index;
+      const page = await this.pdf.getPage(index + 1);
+      if (loadingIndex !== this.data.index) return;
+
+      const viewport = page.getViewport({ scale: 3 });
+      const pw = viewport.width;
+      const ph = viewport.height;
+
+      if (!this.canvas || this.canvas.width !== pw || this.canvas.height !== ph) {
+        if (this.canvas) {
+          const oldCacheKey = `${this.el.object3D.uuid}_${this.canvas.width}_${this.canvas.height}`;
+
+          if (this.mesh.map !== errorTexture && textureCache.has(oldCacheKey)) {
+            textureCache.release(oldCacheKey);
+            this.currentPageTextureIsRetained = false;
+          }
+        }
+
+        this.canvas = document.createElement("canvas");
+        this.canvasContext = this.canvas.getContext("2d");
+        this.canvas.width = pw;
+        this.canvas.height = ph;
+      }
+
+      const renderingSrc = this.data.src;
+      const renderingIndex = index;
+
+      if (this.renderTask) {
+        // If a previous update is rendering, wait for it to finish.
+        await new Promise(resolve => {
+          const interval = setInterval(() => {
+            if (!this.renderTask) {
+              clearInterval(interval);
+              resolve();
+            }
+          }, 100);
+        });
+      }
+
+      this.renderTask = page.render({ canvasContext: this.canvasContext, viewport });
+      await this.renderTask.promise;
+      this.renderTask = null;
+
+      if (this.data.src !== renderingSrc || this.data.index !== renderingIndex) return;
+
+      const cacheKey = this.currentTextureCacheKey();
+      let cacheItem;
+      if (textureCache.has(cacheKey)) {
+        if (this.currentPageTextureIsRetained) {
+          cacheItem = textureCache.get(cacheKey);
+        } else {
+          cacheItem = textureCache.retain(cacheKey);
+        }
+      } else {
+        if (src === "error") {
```

Comment:
Dont think you need to handle this case. Its dont for images since a `media-image` with `src: "error"`is used for displaying errors in general (ex when failing to load a GLTF, or when loading an unknown content type). A `media-pdf` should never get created with`src: "error"`.